Firestore/Protos/build_protos.py

```diff
@@ -286,6 +293,57 @@ def add_copyright(lines):
   return result
 
 
+def nanopb_add_namespaces(lines):
+  """Adds C++ namespaces to the lines.
+
+  Args:
+    lines: The lines to fix.
+
+  Returns:
+    The lines, fixed.
+  """
+  result = []
+  for line in lines:
+    if '@@protoc_insertion_point(includes)' in line:
+      result.append('namespace firebase {\n')
+      result.append('namespace firestore {\n')
+      result.append('\n')
+
+    if '@@protoc_insertion_point(eof)' in line:
+      result.append('}  // namespace firestore\n')
+      result.append('}  // namespace firebase\n')
+      result.append('\n')
+
+    result.append(line)
+
+  return result
+
+
+def nanopb_remove_extern_c(lines):
+  """Removes extern "C" directives from nanopb code.
```

Comment:
Works fine, but seems like it could potentially get us into trouble in the future. Consider if nanopb decided to be a bit friendlier to c++ and generated this for a field called 'delete':

```
struct X {
  #ifdef __cplusplus
  int delete_;
  #else
  int delete;
  #endif
};
```

A better approach might be to look for these complete phrases:
```
#ifdef __cplusplus	
extern "C" {	
#endif
```
and
```
#ifdef __cplusplus
} /* extern "C" */
#endif
```

and then remove them. But then again, nanopb could change the way it generates the comment in the latter case, so that's hardly bullet proof either.

Since the resulting code gets checked in, it gives us a chance to notice if nanopb makes any change along these lines, so we should be ok with what we have now.

Optional.

Reply:
I'm inclined to burn that bridge once we get to it, and not worry too much now. As you say, even investing a bunch of time in making this more robust it ends up being pretty fragile.